patches/tModLoader/Terraria/ModLoader/ModResourcesDisplayStyle.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Terraria.GameContent.UI;
+
+namespace Terraria.ModLoader
+{
+	public abstract class ModResourcesDisplayStyle : ModType, IPlayerResourcesDisplaySet
+	{
+		public int Type { get; internal set; }
+
+		protected sealed override void Register() {
+			ModTypeLookup<ModResourcesDisplayStyle>.Register(this);
+			Type = ResourcesDisplayStyleLoader.Add(this);
+			DisplayName = Mod.GetOrCreateTranslation($"Mods.{Mod.Name}.DisplayStyleName.{Name}");
+		}
+
+		/// <summary>
+		/// The visible name shown in the settings menu. If this isn't overwritten then it defaults to the <see cref="Mod"/>'s display name.
+		/// </summary>
+		public virtual ModTranslation DisplayName {
+			get;
+			internal set;
+		}
+
+		public string SetName => FullName;
```

Comment:
Maybe make this an explicit interface implementation. And just use FullName in context of this type.